models/scan.go

```diff
@@ -0,0 +1,22 @@
+package models
+
+import (
+	"html/template"
+	"net/http"
+	"time"
+
+	"github.com/EFForg/starttls-backend/checker"
+)
+
+// Scan stores the result of a scan of a domain
+type Scan struct {
+	Domain    string               `json:"domain"`    // Input domain
+	Data      checker.DomainResult `json:"scandata"`  // Scan results from starttls-checker
+	Timestamp time.Time            `json:"timestamp"` // Time at which this scan was conducted
+}
+
+// WriteHTML renders a scan result as HTML.
+func (s Scan) WriteHTML(w http.ResponseWriter) error {
+	tmpl := template.Must(template.ParseFiles("views/scan.html.tmpl"))
```

Comment:
Does this file exist? we can also lump this fn into the larger no-JS support PR instead of adding it here

Reply:
:+1: saving for noscript PR